mlir/test/Quantum/AllocationInterfaceTest.mlir

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// RUN: quantum-opt --buffer-deallocation --split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: @existing_deallocation
+func.func @existing_deallocation() {
+    %0 = quantum.alloc( 1) : !quantum.reg
```

Comment:
` %0 = quantum.alloc(1)` or  ` %0 = quantum.alloc( 1)`?  :)

Reply:
Either works for parsing, but unfortunately we have no control over how it is printed because of how the custom syntax declaration works here in the case that two different options are available for the next token after `quantum.alloc(` (we accept both attributes and SSA values).